### PR TITLE
add profile page

### PR DESCRIPTION
**Pull request by TanyaDovzhenko**

### CODE REVIEW

File: src/api/user/update-profile.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+export async function updateProfilePhoto(userId: number, file: any) {
+  const formData = new FormData()
+  formData.append('userId', String(userId))
+  formData.append('photo', file)
+
+  const updatedPhoto: string = await axios.put(
+    `${process.env.REACT_APP_BASE_SERVICE_URL}user-update/photo`, formData)
+    .then((data) => data.data?.photo)
+
+  return updatedPhoto
```
Comment (lerik1408) [inline src/api/user/update-profile.ts:30]: https://github.com/CHI-internship/KraudDonate-Service/pull/4#pullrequestreview-1166008438 - first comment

**Review by lerik1408 — COMMENTED** [review]
Good Job!

File: src/api/user/update-profile.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+export async function updateProfileName(userId: number, name: string) {
+  const updatedName: string = await axios.put(
+    `${process.env.REACT_APP_BASE_SERVICE_URL}user-update/name`,
+    { userId, name })
+    .then((data) => data.data?.name)
+
+  return updatedName
+}
+
+export async function updateProfileLastname(userId: number, lastname: string) {
```
Comment (BillyAleks) [inline src/api/user/update-profile.ts:12]: I have a question for discussion. Why there are two user-update queries???
As for me it looks like it's better in a single query.
But this question is for backend.

File: src/components/profile/profile-avatar/ProfileAvatar.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import style from './ProfileAvatar.module.scss'
+import addIcon from '../../../images/icons/add.icon.svg'
+import { updateProfilePhoto } from '../../../api/user/update-profile'
+
+interface IProfileAvatarProps {
+  userId: number
+  initialAvatar: string | null
+}
+
+export default function ProfileAvatar({ userId, initialAvatar }: IProfileAvatarProps) {
```
Comment (BillyAleks) [inline src/components/profile/profile-avatar/ProfileAvatar.tsx:11]: it should have a type FC<IProfileAvatarProps>

File: src/components/profile/profile-avatar/ProfileAvatar.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        <input ref={ref}
+          type='file'
+          accept='image/*'
+          onChange={(e: any) => { updatePhotoHandler(e.target.files[0]) }}
+          style={{ display: 'none' }}
+        />
+      </div>
+    </div >
+  )
+}
```
Comment (BillyAleks) [inline src/components/profile/profile-avatar/ProfileAvatar.tsx:38]: Please add default props here for component, using prop-types lib

File: src/components/profile/profile-avatar/ProfileAvatar.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import { useState, useRef } from 'react'
+import style from './ProfileAvatar.module.scss'
+import addIcon from '../../../images/icons/add.icon.svg'
+import { updateProfilePhoto } from '../../../api/user/update-profile'
+
+interface IProfileAvatarProps {
+  userId: number
+  initialAvatar: string | null
```
Comment (BillyAleks) [inline src/components/profile/profile-avatar/ProfileAvatar.tsx:8]: I think it's possible to use an optional prop here.

File: src/components/profile/profile-name/ProfileName.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+import { updateProfileLastname, updateProfileName } from '../../../api/user/update-profile'
+
+
+interface IProfileNameProps {
+  userId: number
+  initialName: string
+  initialLastname: string
+}
+
+export default function ProfileName({ userId, initialName, initialLastname }: IProfileNameProps) {
```
Comment (BillyAleks) [inline src/components/profile/profile-name/ProfileName.tsx:18]: same about component type

**Review by BillyAleks — CHANGES_REQUESTED** [review]
Thank you. Please check comments.
Also would be great to add to each components folder index.ts file to make further imports shorter.

File: src/api/user/update-profile-name.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+interface IUpdateName {
+  userId: number
+  name: string
+  lastname: string
+}
+
+export async function updateName({ userId, name, lastname }: IUpdateName) {
+  const updatedName = await axios.put(
+    `${process.env.REACT_APP_BASE_SERVICE_URL}user-update/name`,
```
Comment (lerik1408) [inline src/api/user/update-profile-name.ts:11]: 👀

File: src/api/user/update-profile-name.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import axios from 'axios'
+
+interface IUpdateName {
+  userId: number
+  name: string
+  lastname: string
+}
+
+export async function updateName({ userId, name, lastname }: IUpdateName) {
+  const updatedName = await axios.put(
```
Comment (lerik1408) [inline src/api/user/update-profile-name.ts:10]: const instance = axios.create({
  baseURL: 'https://some-domain.com/api/',
  timeout: 1000,
  headers: {'X-Custom-Header': 'foobar'}
});

File: src/api/user/update-profile.ts [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+  name: string
+  lastname: string
+}
+
+export async function updatePofile({ userId, name, lastname }: IUpdateProfile) {
+  const updatedUser = await axiosInstance.patch(
+    'user', { userId, name, lastname })
+    .then(data => data.data)
+
+  return updatedUser
```
Comment (lerik1408) [inline src/api/user/update-profile.ts:14]: 👀

**Review by lerik1408 — COMMENTED** [review]
GJ

**Review by lerik1408 — COMMENTED** [review]
GJ

File: package.json [hunk 10]
```diff
@@ -36,16 +50,23 @@
...
     "@testing-library/react": "^13.4.0",
     "@testing-library/user-event": "^13.5.0",
     "@types/jest": "^27.5.2",
+<<<<<<< HEAD
+=======
     "@types/node": "^16.18.2",
     "@types/react": "^18.0.23",
     "@types/react-dom": "^18.0.7",
     "@types/react-router-dom": "^5.3.3",
+>>>>>>> main
```
Comment (lerik1408) [inline package.json:69]: Something with merge

**Review by lerik1408 — COMMENTED** [review]
GJ

**Pull request merged**